### PR TITLE
feat(cli): add inline suppression comments for findings

### PR DESCRIPTION
**Pull request by Alberto-Codes**
Adds `# docvet: ignore[rule]` and `# docvet: ignore-file[rule]` inline suppression comments, letting developers acknowledge known deviations without disabling entire rules or inflating the noise floor. Suppression operates as a post-filter on the findings list — no check functions are modified.

- Add `_suppression.py` module with tokenize-based comment parser and finding filter
- Wire suppression into `_output_and_exit()` before flattening, so both output and exit code exclude suppressed findings
- Add verbose suppressed listing to stderr and `"suppressed"` array to JSON format
- Add `scaffold-incomplete` rule to MCP catalog (was missing)
- Add dedicated `suppression.md` docs page and inline suppression section to CLI reference
- Add 48 tests (43 unit + 5 integration CLI roundtrips)

fix(cli): prevent uppercase rule IDs from degrading to blanket suppression

Widen `_DIRECTIVE_RE` bracket character class from `[a-z0-9, -]` to `[^\]]+` so non-standard input (uppercase, underscores) is captured and forwarded to `_parse_rules` for validation, rather than silently falling back to blanket suppression. Add 4 regression tests.

Test: `uv run pytest tests/unit/test_cli_suppression.py tests/integration/test_suppression_cli.py -v`

---

### Checklist
- [x] Self-reviewed my code
- [x] Tests pass (`uv run pytest`)
- [x] Lint passes (`uv run ruff check .`)
- [x] Types pass (`uv run ty check`)
- [x] Breaking changes use `!` in title and `BREAKING CHANGE:` in body

### Review Focus
- `_suppression.py`: regex correctness, tokenize edge cases, `_is_suppressed` None-vs-absent dict logic
- `_output.py`: suppression wired before flatten loop, source file caching
- Regression tests for H1 (uppercase rule → blanket suppression bug, now fixed)

### Related
- Story: `_bmad-output/implementation-artifacts/32-4-inline-suppression-comments.md`
- Follow-up: #392 (reporting.py split — 518 > 500-line gate)

### CODE REVIEW

File: docs/site/cli-reference.md [hunk 1]
```diff
@@ -46,8 +46,9 @@ The `--format json` option produces a structured JSON object for programmatic co
 ```
 
 - Each finding has seven fields: `file`, `line`, `symbol`, `rule`, `message`, `category`, `severity`.
-- The `severity` field is derived from `category`: `"required"` maps to `"high"`, `"recommended"` maps to `"low"`. It has exactly two values and is a convenience alias, not a separate signal.
-- The `summary` object includes `total`, `by_category` (with `required` and `recommended` counts), and `files_checked`.
+- The `severity` field is derived from `category`: `"required"` maps to `"high"`, `"scaffold"` maps to `"medium"`, `"recommended"` maps to `"low"`. It is a convenience alias, not a separate signal.
+- The `summary` object includes `total`, `by_category` (with `required`, `recommended`, and `scaffold` counts), and `files_checked`.
+- A `suppressed` array is included containing findings that were suppressed by inline `# docvet: ignore` comments. Each entry has the same seven fields as a finding.
 - Whitespace and indentation are not part of the schema contract — always parse with a JSON parser.
 - Exit codes are unchanged: `0` for no findings, `1` when findings exist in a `fail_on` check.
```
[the remark this answers is not in the document]
Comment (Alberto-Codes) [reply docs/site/cli-reference.md]: Addressed in latest push — updated the JSON example to include `scaffold: 0` in `by_category` and a `suppressed` array with a sample entry.
